taskRunner/multi_stage.py

```diff
@@ -110,145 +70,172 @@ def main():
+            with open("{0}/processors.csv".format(workspace_directory), 'a', newline='') as csvfile:
                 writer = csv.writer(csvfile)
-                data = [[integration_id, taskId, log_group_name, log_stream_name, application_uuid, container_name, application_type]]
+                data = [[integration_id, task_id, log_group_name, log_stream_name, application_uuid, container_name, application_type]]
 
                 for row in data:
                     writer.writerow(row)
                 csvfile.close()
-            
-            # sync
-            sts_client = boto3_client("sts")
-            account_id = sts_client.get_caller_identity()["Account"]
-            bucket_name = "tfstate-{0}".format(account_id)
-            prefix = "{0}/logs/{1}".format(env,integration_id)
-
-            try:
-                output = subprocess.run(["aws", "s3", "sync", workspaceDir, "s3://{0}/{1}/".format(bucket_name, prefix)]) 
-                logger.info(output)
-            except subprocess.CalledProcessError as e:
-                logger.info(f"command failed with return code {e.returncode}")
-            
-            waiter = ecs_client.get_waiter('tasks_stopped')
-            waiter.wait(
-                cluster=cluster_name,
-                tasks=[task_arn],
-                WaiterConfig={
-                    'Delay': 30,
-                    'MaxAttempts': 2000
-                }
-            )
 
-            response = ecs_client.describe_tasks(
-                cluster=cluster_name,
-                tasks=[task_arn]
-            )
+            sync_logs(sts_client, config, integration_id, workspace_directory)
+            exit_code = poll_task(ecs_client, config, task_arn)
 
-            exit_code = response['tasks'][0]['containers'][0]['exitCode']
-            
+            now = datetime.now(timezone.utc).timestamp()
             if exit_code == 0:
-                logger.info("success: container_name={0},application_type={1}".format(container_name, application_type))
+                workflow_instance_client.put_workflow_instance_status(integration_id, application_uuid, 'SUCCEEDED', now, session_token)
+                logger.info("success: container_name={0}, application_type={1}".format(container_name, application_type))
             else:
-                logger.error("error: container_name={0},application_type={1}".format(container_name, application_type))
+                workflow_instance_client.put_workflow_instance_status(integration_id, application_uuid, 'FAILED', now, session_token)
+                logger.error("error: container_name={0}, application_type={1}".format(container_name, application_type))
                 sys.exit(1)
 
-            logger.info("fargate task has stopped: " + task_definition_name)
+    logger.info("fargate task has stopped: " + task_definition_name)
+
+def start_task(ecs_client, config, task_definition_name, container_name, command):
+    if config.IS_LOCAL:
```

Comment:
the reason I put these conditionals at the top of each function was to make sure that the function calls themselves were tested locally as well and the only code that wasn't tested were the literal calls to AWS (as was the existing reason why all the code was split up in the first place I believe)